lib/dry/auto_inject/builder.rb

```diff
@@ -0,0 +1,31 @@
+require 'dry/auto_inject/strategies'
+require 'dry/auto_inject/injector'
+
+module Dry
+  module AutoInject
+    class Builder
+      # @api private
+      attr_reader :container
+
+      # @api private
+      attr_reader :strategies
+
+      def initialize(container, options = {})
+        @container = container
+        @strategies = options.fetch(:strategies) { Strategies }
+
+        strategies.keys.each do |strategy_name|
```

Comment:
This is a `Dry::Container`, not a regular `Hash`, so all I've got available to me from the public API is `#keys`.